docs/conf.py

```diff
@@ -176,13 +176,21 @@
 html_sidebars = {"**": ["sidebar-nav-bs.html", "sidebar-ethical-ads.html"]}
 
 # For edit button
-html_context = {
-    "github_user": "pybamm-team",
-    "github_repo": "pybamm",
-    "github_version": "develop",
-    "doc_path": "docs/",
-}
+html_context.update(
+    {
+        "github_user": "pybamm-team",
+        "github_repo": "pybamm",
+        "github_version": "develop",
+        "doc_path": "docs/",
+    }
+)
+
+# Set canonical URL from the Read the Docs Domain
+html_baseurl = os.environ.get("READTHEDOCS_CANONICAL_URL", "")
```

Comment:
```suggestion
html_baseurl = os.getenv("READTHEDOCS_CANONICAL_URL", "")
```

If we decide to use `os.getenv` below, we should use the same here too. Makes it a little more readable.

Missed this one in the last review.

Reply:
I would have liked to push back on these suggestions since readability shouldn't be a concern with an API difference, and they mean the same thing, see here:

https://github.com/python/cpython/blob/ef21e48cac548025aa76704b84b489f0c111e1de/Lib/os.py#L807-L813

but I won't wish to hold up the PR :)

Reply:
Applied the suggestion